InlineBeginTransactionWithoutExecutorTest is flaky

```
[INFO] Running com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithoutExecutorTest
Error:  Tests run: 56, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.478 s <<< FAILURE! -- in com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithoutExecutorTest
Error:  com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithoutExecutorTest.testQueryAsyncWithInlineBeginDidNotReturnTransaction -- Time elapsed: 0.030 s <<< FAILURE!
java.lang.AssertionError: expected com.google.cloud.spanner.SpannerException to be thrown, but nothing was thrown
	at org.junit.Assert.assertThrows(Assert.java:1028)
	at org.junit.Assert.assertThrows(Assert.java:981)
	at com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithoutExecutorTest.testQueryAsyncWithInlineBeginDidNotReturnTransaction(InlineBeginTransactionTest.java:1889)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:49)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:120)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:95)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:69)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

[INFO] Running com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithExecutorTest
[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.678 s -- in com.google.cloud.spanner.InlineBeginTransactionTest$InlineBeginTransactionWithExecutorTest

```

--- a/google-cloud-spanner/src/test/java/com/google/cloud/spanner/InlineBeginTransactionTest.java
+++ b/google-cloud-spanner/src/test/java/com/google/cloud/spanner/InlineBeginTransactionTest.java
@@ -1668,192 +1668,203 @@ public class InlineBeginTransactionTest {
       assertEquals(0, countRequests(CommitRequest.class));
     }
 
+    static void runWithIgnoreInlineBegin(Runnable runnable) {
+      // This will cause statements that requests a transaction to not return a transaction id.
+      mockSpanner.setIgnoreInlineBeginRequest(true);
+      try {
+        runnable.run();
+      } finally {
+        mockSpanner.setIgnoreInlineBeginRequest(false);
+      }
+    }
+
     @Test
     public void testQueryWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction -> {
-                            try (ResultSet rs =
-                                transaction.executeQuery(SELECT1_UNION_ALL_SELECT2)) {
-                              while (rs.next()) {}
-                            }
-                            return null;
-                          }));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction -> {
+                                  try (ResultSet rs =
+                                      transaction.executeQuery(SELECT1_UNION_ALL_SELECT2)) {
+                                    while (rs.next()) {}
+                                  }
+                                  return null;
+                                }));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testReadWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction ->
-                              transaction.readRow(
-                                  "FOO", Key.of(1L), Collections.singletonList("BAR"))));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ReadRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction ->
+                                    transaction.readRow(
+                                        "FOO", Key.of(1L), Collections.singletonList("BAR"))));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ReadRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testUpdateWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(transaction -> transaction.executeUpdate(UPDATE_STATEMENT)));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(transaction -> transaction.executeUpdate(UPDATE_STATEMENT)));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testBatchUpdateWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction -> {
-                            transaction.batchUpdate(Collections.singletonList(UPDATE_STATEMENT));
-                            return null;
-                          }));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteBatchDmlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction -> {
+                                  transaction.batchUpdate(
+                                      Collections.singletonList(UPDATE_STATEMENT));
+                                  return null;
+                                }));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteBatchDmlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testQueryAsyncWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      final ExecutorService executor = Executors.newSingleThreadExecutor();
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException outerException =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction -> {
-                            try (AsyncResultSet rs =
-                                transaction.executeQueryAsync(SELECT1_UNION_ALL_SELECT2)) {
-                              return SpannerApiFutures.get(
-                                  rs.setCallback(
-                                      executor,
-                                      resultSet -> {
-                                        try {
-                                          while (true) {
-                                            switch (resultSet.tryNext()) {
-                                              case OK:
-                                                break;
-                                              case DONE:
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            final ExecutorService executor = Executors.newSingleThreadExecutor();
+            SpannerException outerException =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction -> {
+                                  try (AsyncResultSet rs =
+                                      transaction.executeQueryAsync(SELECT1_UNION_ALL_SELECT2)) {
+                                    return SpannerApiFutures.get(
+                                        rs.setCallback(
+                                            executor,
+                                            resultSet -> {
+                                              try {
+                                                while (true) {
+                                                  switch (resultSet.tryNext()) {
+                                                    case OK:
+                                                      break;
+                                                    case DONE:
+                                                      return CallbackResponse.DONE;
+                                                    case NOT_READY:
+                                                      return CallbackResponse.CONTINUE;
+                                                  }
+                                                }
+                                              } catch (SpannerException e) {
                                                 return CallbackResponse.DONE;
-                                              case NOT_READY:
-                                                return CallbackResponse.CONTINUE;
-                                            }
-                                          }
-                                        } catch (SpannerException e) {
-                                          return CallbackResponse.DONE;
-                                        }
-                                      }));
-                            }
-                          }));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, outerException.getErrorCode());
-      assertThat(outerException.getMessage())
-          .contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+                                              }
+                                            }));
+                                  }
+                                }));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, outerException.getErrorCode());
+            assertThat(outerException.getMessage())
+                .contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
 
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testUpdateAsyncWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction ->
-                              SpannerApiFutures.get(
-                                  transaction.executeUpdateAsync(UPDATE_STATEMENT))));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction ->
+                                    SpannerApiFutures.get(
+                                        transaction.executeUpdateAsync(UPDATE_STATEMENT))));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteSqlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test
     public void testBatchUpdateAsyncWithInlineBeginDidNotReturnTransaction() {
-      DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
-      // This will cause the first statement that requests a transaction to not return a transaction
-      // id.
-      mockSpanner.ignoreNextInlineBeginRequest();
-      SpannerException e =
-          assertThrows(
-              SpannerException.class,
-              () ->
-                  client
-                      .readWriteTransaction()
-                      .run(
-                          transaction ->
-                              SpannerApiFutures.get(
-                                  transaction.batchUpdateAsync(
-                                      Collections.singletonList(UPDATE_STATEMENT)))));
-      assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
-      assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
-      assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
-      assertThat(countRequests(ExecuteBatchDmlRequest.class)).isEqualTo(1);
-      assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+      runWithIgnoreInlineBegin(
+          () -> {
+            DatabaseClient client = spanner.getDatabaseClient(DatabaseId.of("p", "i", "d"));
+            SpannerException e =
+                assertThrows(
+                    SpannerException.class,
+                    () ->
+                        client
+                            .readWriteTransaction()
+                            .run(
+                                transaction ->
+                                    SpannerApiFutures.get(
+                                        transaction.batchUpdateAsync(
+                                            Collections.singletonList(UPDATE_STATEMENT)))));
+            assertEquals(ErrorCode.FAILED_PRECONDITION, e.getErrorCode());
+            assertThat(e.getMessage()).contains(AbstractReadContext.NO_TRANSACTION_RETURNED_MSG);
+            assertThat(countRequests(BeginTransactionRequest.class)).isEqualTo(0);
+            assertThat(countRequests(ExecuteBatchDmlRequest.class)).isEqualTo(1);
+            assertThat(countRequests(CommitRequest.class)).isEqualTo(0);
+          });
     }
 
     @Test

--- a/google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java
+++ b/google-cloud-spanner/src/test/java/com/google/cloud/spanner/MockSpannerServiceImpl.java
@@ -607,7 +607,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
   private ConcurrentMap<ByteString, Boolean> commitRetryTransactions = new ConcurrentHashMap<>();
   private final AtomicBoolean abortNextTransaction = new AtomicBoolean();
   private final AtomicBoolean abortNextStatement = new AtomicBoolean();
-  private final AtomicBoolean ignoreNextInlineBeginRequest = new AtomicBoolean();
+  private final AtomicBoolean ignoreInlineBeginRequest = new AtomicBoolean();
   private ConcurrentMap<String, AtomicLong> transactionCounters = new ConcurrentHashMap<>();
   private ConcurrentMap<String, List<ByteString>> partitionTokens = new ConcurrentHashMap<>();
   private ConcurrentMap<ByteString, Instant> transactionLastUsed = new ConcurrentHashMap<>();
@@ -790,8 +790,8 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
     }
   }
 
-  public void ignoreNextInlineBeginRequest() {
-    ignoreNextInlineBeginRequest.set(true);
+  public void setIgnoreInlineBeginRequest(boolean ignore) {
+    ignoreInlineBeginRequest.set(ignore);
   }
 
   public void freeze() {
@@ -1063,7 +1063,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
                     .setMetadata(
                         ResultSetMetadata.newBuilder()
                             .setTransaction(
-                                ignoreNextInlineBeginRequest.getAndSet(false)
+                                ignoreInlineBeginRequest.get()
                                     ? Transaction.getDefaultInstance()
                                     : Transaction.newBuilder().setId(transactionId).build())
                             .build());
@@ -1096,7 +1096,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
           metadata
               .toBuilder()
               .setTransaction(
-                  ignoreNextInlineBeginRequest.getAndSet(false)
+                  ignoreInlineBeginRequest.get()
                       ? Transaction.getDefaultInstance()
                       : Transaction.newBuilder().setId(transactionId).build())
               .build();
@@ -1197,7 +1197,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
                 .setMetadata(
                     ResultSetMetadata.newBuilder()
                         .setTransaction(
-                            ignoreNextInlineBeginRequest.getAndSet(false)
+                            ignoreInlineBeginRequest.get()
                                 ? Transaction.getDefaultInstance()
                                 : Transaction.newBuilder().setId(transactionId).build())
                         .build())
@@ -1730,7 +1730,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
           metadata
               .toBuilder()
               .setTransaction(
-                  ignoreNextInlineBeginRequest.getAndSet(false)
+                  ignoreInlineBeginRequest.get()
                       ? Transaction.getDefaultInstance()
                       : Transaction.newBuilder().setId(transactionId).build())
               .build();
@@ -1785,7 +1785,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
                   ResultSetMetadata.newBuilder()
                       .setRowType(StructType.newBuilder().addFields(field).build())
                       .setTransaction(
-                          ignoreNextInlineBeginRequest.getAndSet(false)
+                          ignoreInlineBeginRequest.get()
                               ? Transaction.getDefaultInstance()
                               : Transaction.newBuilder().setId(transaction.getId()).build())
                       .build())
@@ -1798,7 +1798,7 @@ public class MockSpannerServiceImpl extends SpannerImplBase implements MockGrpcS
                   ResultSetMetadata.newBuilder()
                       .setRowType(StructType.newBuilder().addFields(field).build())
                       .setTransaction(
-                          ignoreNextInlineBeginRequest.getAndSet(false)
+                          ignoreInlineBeginRequest.get()
                               ? Transaction.getDefaultInstance()
                               : Transaction.newBuilder().setId(transaction.getId()).build())
                       .build())

